### PR TITLE
feat(ci): s1ap integ tests against dockerized agw

### LINKED ISSUE #13695: CI support for dockerized agw attach dettach test
### Background:

In [#13475](https://github.com/magma/magma/issues/13475) we analyzed what needs to be done to get the s1ap running with the dockerized agw. This story already enabled one s1ap test (attach/dettach). #13684 will enable more tests as well as improve the setup. This ticket will only enable the attach/dettach test

### Acceptance criteria:

- Add a ci job for running the s1ap attach/dettach against the dockerized agw

### Reference

- Follow up ticket [#14123](https://github.com/magma/magma/issues/14123)

### PR DESCRIPTION
**Pull request by alexzurbonsen**
## Summary

- Contributes to https://github.com/magma/magma/issues/13695
- A new workflow is created, which runs some integration tests against a containerized instance of AGW running on the Magma Dev VM.
  - *Only attach-detach test* for this PR. 
- This is the first iteration of testing the containerized AGW in CI.
  - The workflow was designed based on the exiting workflows.
  - It is expected that further optimisation can be made in follow ups later on.
  - Some integration test are flaky in general, such that for this stage, _one_ successful run will be seen as solution.
- The tests are run against the `latest` images of the services available.
  - Especially, it is out of scope to build the containerized AGW within this effort and PR.
  - Using the latest available images should provide sufficient additional information, if there is a sensible trigger to the workflow. In particular, any edge cases or race conditions are acquiesced!

## Test Plan

- See runs: https://github.com/magma/magma/actions/workflows/lte-integ-test-containerized.yml
  -  :heavy_check_mark: Successful at https://github.com/magma/magma/actions/runs/3190270518/jobs/5205143748.

## Additional Information

- As a guide: https://github.com/magma/magma/blob/master/lte/gateway/docker/README.md#running-the-containerized-agw-locally-on-the-magma-vm

### CODE REVIEW

File: lte/gateway/fabfile.py
Comment (MoritzThomasHuebner) [inline lte/gateway/fabfile.py]: We should use `gateway_ip is None` here:

https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none

Alternatively, check with `if not gateway_ip`, that way we also set the `gateway_ip` if it is an empty string.

Comment (alexzurbonsen) [reply lte/gateway/fabfile.py]: This change is originally from [this PR](https://github.com/magma/magma/pull/13842) and will disappear from the diff once it is merged. Can you also leave a comment in there please?

Comment (MoritzThomasHuebner) [reply lte/gateway/fabfile.py]: i copied it over

File: lte/gateway/fabfile.py
Comment (Neudrino) [inline lte/gateway/fabfile.py:542]: As a first step only the `s1aptests/test_attach_detach.py` is run. More will follow in further PRs.

File: lte/gateway/fabfile.py
Comment (Neudrino) [inline lte/gateway/fabfile.py:556]: We are aware that this needs to be replaced by a better health check, however, this might either be solved by a `docker-compose` upgrade or helper functionality currently still under development.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (jheidbrink) [inline .github/workflows/lte-integ-test-containerized.yml]: Shouldn't this be `docker-ci.artifactory...`?

**Review by jheidbrink — APPROVED** [review]

File: lte/gateway/fabfile.py
Comment (Neudrino) [reply lte/gateway/fabfile.py:556]: This will be fixed by a proper health check in the follow up PR with https://github.com/magma/magma/pull/14090/commits/11fbfe9f6686435fb6e280cc7ad6aa136b09e6a2.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml]: Valid question. The options are: `docker` (=docker-prod), `docker-ci`, `docker-test`.

For creating this workflow I was seeing the prod artifactory to be most suitable, as it would contain the most stable artifacts, such as to avoid any failures due to broken artifacts.
In the long term it should probably take artifacts from the repository, where the master branch build all job pushes to.

Let's discuss orally and I will create a follow up PR (or integrate it into https://github.com/magma/magma/pull/14090/) to reflect the outcome of the discussion. :slightly_smiling_face:

Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml]: :heavy_check_mark: Changed.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (sebathomas) [inline .github/workflows/lte-integ-test-containerized.yml:19]: `'v1.*'` branches missing

File: .github/workflows/lte-integ-test-containerized.yml
Comment (voisey) [inline .github/workflows/lte-integ-test-containerized.yml:12]: I don't think the AGW here is necessary seeing as we don't have it for the other LTE integ test workflows.

File: lte/gateway/fabfile.py
Comment (voisey) [inline lte/gateway/fabfile.py]: Setup -> Set up

File: lte/gateway/fabfile.py
Comment (voisey) [inline lte/gateway/fabfile.py]: Setup -> Set up

File: lte/gateway/fabfile.py
Comment (voisey) [inline lte/gateway/fabfile.py]: It would be helpful if it mentioned the containerized AGW in the docstring, i.e. how this is different to the other integ_test functions.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (sebathomas) [inline .github/workflows/lte-integ-test-containerized.yml:65]: Why not upload the test results as in the other PR?
https://github.com/magma/magma/pull/14023/files#diff-3386839a8fdf9c48220ffbd4a015014d7be1ebc14a5792f8b63029f00ce725ecR81

File: lte/gateway/fabfile.py
Comment (voisey) [inline lte/gateway/fabfile.py:529]: When this gets merged, please add the removal of this to the scope of that issue so we don't forget it.

File: lte/gateway/fabfile.py
Comment (voisey) [inline lte/gateway/fabfile.py]: I would suggest mentioning the containerization in the docstring, i.e. minimally "Starts the gateway" -> "Starts the containerized gateway".

File: .github/workflows/lte-integ-test-containerized.yml
Comment (voisey) [inline .github/workflows/lte-integ-test-containerized.yml]: There aren't spaces between the first three blocks but there are between the rest of them. Can you please make it consistent? I would suggest removing the spaces for consistency with the other workflows.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml:19]: No, this is intentional and correct. As for the Bazel integration test workflow, this makes currently no sense to activate for older release branches.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml:65]: This is not in scope of this first PR as it is essentially unnecessary to complete the story. And it would really only make sense if more than the attach-detach-test would be executed.
We can think about the reporting in a follow up though.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml:12]: But it does not hurt either.

File: lte/gateway/fabfile.py
Comment (Neudrino) [reply lte/gateway/fabfile.py:529]: Added to https://github.com/magma/magma/issues/13912.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml]: They were originally introduced for logic division of different aspects of the workflow. I removed them anyway.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (voisey) [inline .github/workflows/lte-integ-test-containerized.yml]: Why is the fabric version printed here?

File: .github/workflows/lte-integ-test-containerized.yml
Comment (sebathomas) [reply .github/workflows/lte-integ-test-containerized.yml:19]: It is essentially a smoke test for the dockerized agw and the only one we have. It absolutely makes sense to run for release branches. There is nothing else that will show problems with the setup.

Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml:19]: While it might make sense to run it for release branches *in future*, currently, I cannot see any sense in it, as it will most likely not work without substantial amount of backporting on v1.8.
Additionally, I see this clearly out of scope of the ticket, such that it will make no sense at all to add it to this PR. A new story could be created if this is desired in future.

Comment (Neudrino) [reply .github/workflows/lte-integ-test-containerized.yml:19]: On second thought, it does not make sense at all to add the versioned branches here, as the workflow is not designed to work with them anyway.
To generalise the workflow and make it work with any branch is also clearly out of scope of the ticket.

**Review by voisey — APPROVED** [review]
Lgtm but please wait for @sebathomas to approve before merging.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (sebathomas) [reply .github/workflows/lte-integ-test-containerized.yml:19]: It was one of the few differences between this and the other version of the PR in #14023. Just trying to avoid things getting lost.

But I see that it probably makes little sense while we always pick the latest image version. Added it to #14052.

**Review by sebathomas — APPROVED** [review]

**Pull request merged**
